server/src/main/java/org/elasticsearch/transport/RequestHandlerRegistry.java

```diff
@@ -118,15 +132,34 @@ public static <R extends TransportRequest> RequestHandlerRegistry<R> replaceHand
     }
 
     public void addRequestStats(int messageSize) {
-        statsTracker.addRequestStats(messageSize);
+        statsTracker().addRequestStats(messageSize);
     }
 
     @Override
     public void addResponseStats(int messageSize) {
-        statsTracker.addResponseStats(messageSize);
+        statsTracker().addResponseStats(messageSize);
     }
 
     public TransportActionStats getStats() {
+        var statsTracker = existingStatsTracker();
+        if (statsTracker == null) {
+            return TransportActionStats.EMPTY;
+        }
         return statsTracker.getStats();
     }
+
+    private TransportActionStatsTracker statsTracker() {
+        var tracker = existingStatsTracker();
```

Comment:
If you're looking for yet-weaker memory semantics, I think this one can be a regular (nonvolatile) read. If it's null then the `compareAndExchange` will establish the right guarantees, and if it's not null then it has to be the right value.

Risks perhaps creating a few more unused `TransportActionStatsTracker` instances tho I guess.

Reply:
Yea somehow a plain read in this kind of spot weirds me out a little :D but you are right, we could relax these spots further technically I guess I just don't feel adventurous enough for that this evening (I do realise though that we actually do this kind of thing ... sync write -> plain read ... in a couple spots like the CS + routing table computation already).